dcql_query & presentation_definition are not strings (#422)
openid/OpenID4VP: openid-4-verifiable-presentations-1_0.md

This specification defines the following new request parameters:

`presentation_definition`:

> : A JSON object containing a Presentation Definition. See (#request_presentation_definition) for more details.

Reviewer: I don't think this change is helpful. This should clarify that the encoding varies between the query string (where it is a JSON-encoded string) and the request object and DC API call (where it is an object and it is not additionally JSON-encoded).

Reviewer 2: I sort of understand this perspective @danielfett, however I'm struggling to see what other interpretation a developer might have about how to convey a JSON object in a query parameter value?

Author: @danielfett Are we at least agreed the current text is bad? It has resulted in people doing this in request objects:

`"dcql_query": "{\"id\":\"ffc717a3-abaf-4ec3-9c55-a9b8e998874c\",\"name\":\"A`

The way I've done it here (combined with the text below explaining how to put JSON objects into the url query) seems consistent with how it's done in RAR ( https://datatracker.ietf.org/doc/html/rfc9396#section-2 ), and we have the example to try and help people struggling to interpret it too.

Reviewer: When this is a DC API call, this is an object (not JSON-encoded).
```suggestion
: An object containing a Presentation Definition. See (#request_presentation_definition) for more details.
```

Author: If we're going to make that change for that reason I suggest doing it consistently everywhere we use the term 'JSON object' in a separate PR.

`presentation_definition_uri`:
: A string containing an HTTPS URL pointing to a resource where a Presentation Definition JSON object can be retrieved. See (#request_presentation_definition_uri) for more details.

`dcql_query`:

> : A JSON object containing a DCQL query as defined in (#dcql_query).

Reviewer: As above
```suggestion
: An object containing a DCQL query as defined in (#dcql_query).
```

Exactly one of the following parameters MUST be present in the Authorization Request: `dcql_query`, `presentation_definition`, `presentation_definition_uri`, or a `scope` value representing a Presentation Definition.

> In the context of an authorization request according to [RFC6749], parameters containing JSON objects are encoded using the application/x-www-form-urlencoded format of the serialized JSON.

Reviewer: ```suggestion
In the context of an authorization request according to [@RFC6749], parameters containing JSON objects are encoded using the application/x-www-form-urlencoded format of the serialized JSON.
```

> In the context of an authorization request according to [@RFC6749], parameters containing JSON objects are encoded using the application/x-www-form-urlencoded format of the serialized JSON.

Reviewer: ```suggestion
In the context of an authorization request according to [@RFC6749], parameters containing objects are transferred as JSON-serialized strings (using the application/x-www-form-urlencoded format as usual for request parameters).
```
